### PR TITLE
Support Vertical Mixed Text

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp.Drawing/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Updates the glyph renderers to handle support for [Unicode Standard Annex #50 Unicode Vertical Text Layout](https://www.unicode.org/reports/tr50/) 

Note: I've discovered an issue when rendering the decorations over certain CFF fonts in that holes are left, however I'm unable to determine what the cause could be. The paths themselves look ok. 

This only affects filling a glyph collection.

### CODE REVIEW

File: src/ImageSharp.Drawing/Shapes/Text/BaseGlyphBuilder.cs [hunk 1]
```diff
@@ -147,30 +147,32 @@ public virtual void SetDecoration(TextDecorations textDecorations, Vector2 start
...
+            Vector2 c = end + pad;
+            Vector2 d = end - pad;
 
             // Drawing is always centered around the point so we need to offset by half.
             Vector2 offset = Vector2.Zero;
             if (textDecorations == TextDecorations.Overline)
             {
                 // CSS overline is drawn above the position, so we need to move it up.
-                offset = new(0, -(thickness * .5F));
+                offset = rotated ? new(thickness * .5F, 0) : new(0, -(thickness * .5F));
```
Comment (tocsoft) [inline src/ImageSharp.Drawing/Shapes/Text/BaseGlyphBuilder.cs:163]: do we still need these offsets?, now we seem to also be applying them above? 

Reason I ask is looking at the onion skin of the changed test images shows the decorations have shifted and are less sharp (i.e. they look like they are .5 of a pixel off and thus aliased edges are happening)

Comment (tocsoft) [reply src/ImageSharp.Drawing/Shapes/Text/BaseGlyphBuilder.cs:163]: fair enough...just figured i'd check. I'm sure we can circle back round to the minor visual differences, and as you say try and tackle that at the rasterizer level, which will improve things acros the board not just for test rendering.

**Review by tocsoft — APPROVED** [review]

**Comment by JimBobSquarePants** [comment]
@tocsoft I couldn't help myself and implemented the smart pixel clamping I was on about in a manner that allowed preserving defined pen thickness. The results look great 😁

**Review by tocsoft — APPROVED** [review]
looks way better all around 👍

**Pull request merged**